File: app/src/main/java/ru/mamykin/foreignbooksreader/di/AppComponent.java
package ru.mamykin.foreignbooksreader.di;

import javax.inject.Singleton;

import dagger.Component;
import ru.mamykin.foreignbooksreader.ReaderApp;
import ru.mamykin.foreignbooksreader.di.modules.ApiModule;
import ru.mamykin.foreignbooksreader.di.modules.AppModule;
import ru.mamykin.foreignbooksreader.di.modules.DatabaseModule;
import ru.mamykin.foreignbooksreader.di.modules.MappersModule;
import ru.mamykin.foreignbooksreader.di.modules.PreferencesModule;
import ru.mamykin.foreignbooksreader.presenters.AboutPresenter;
import ru.mamykin.foreignbooksreader.presenters.BookDetailsPresenter;
import ru.mamykin.foreignbooksreader.presenters.DeviceBooksPresenter;
import ru.mamykin.foreignbooksreader.presenters.DropboxBooksPresenter;
import ru.mamykin.foreignbooksreader.presenters.MainPresenter;
import ru.mamykin.foreignbooksreader.presenters.StorePresenter;
import ru.mamykin.foreignbooksreader.presenters.MyBooksPresenter;
import ru.mamykin.foreignbooksreader.presenters.ReadBookPresenter;
import ru.mamykin.foreignbooksreader.presenters.SettingsPresenter;
import ru.mamykin.foreignbooksreader.ui.activities.BaseActivity;
import ru.mamykin.foreignbooksreader.ui.adapters.BooksRecyclerAdapter;
import ru.mamykin.foreignbooksreader.ui.adapters.BooksStoreRecyclerAdapter;
import ru.mamykin.foreignbooksreader.ui.adapters.DropboxRecyclerAdapter;
import ru.mamykin.foreignbooksreader.ui.adapters.FilesRecyclerAdapter;
import ru.mamykin.foreignbooksreader.ui.adapters.PromotionsRecyclerAdapter;
import ru.mamykin.foreignbooksreader.ui.fragments.MyBooksFragment;
import ru.mamykin.foreignbooksreader.preferences.PreferencesManager;

/**
 * Creation date: 5/29/2017
 * Creation time: 11:39 AM
 * @author Andrey Mamykin(mamykin_av)
 */
@Component(modules = {AppModule.class, ApiModule.class, DatabaseModule.class, PreferencesModule.class, MappersModule.class})
@Singleton
public interface AppComponent {
    void inject(MyBooksFragment fragment);

    void inject(BaseActivity activity);

    void inject(FilesRecyclerAdapter adapter);

    void inject(DropboxRecyclerAdapter adapter);

    void inject(BooksRecyclerAdapter adapter);

    void inject(BooksStoreRecyclerAdapter adapter);

    void inject(PromotionsRecyclerAdapter adapter);

    void inject(PreferencesManager manager);

    void inject(BookDetailsPresenter presenter);

    void inject(DropboxBooksPresenter presenter);

    void inject(MyBooksPresenter presenter);

    void inject(SettingsPresenter presenter);

    void inject(ReadBookPresenter presenter);

    void inject(StorePresenter presenter);

    void inject(AboutPresenter presenter);

    void inject(DeviceBooksPresenter presenter);

    void inject(MainPresenter presenter);

    void inject(ReaderApp application);
}

File: app/src/main/java/ru/mamykin/foreignbooksreader/di/AppComponent.kt
package ru.mamykin.foreignbooksreader.di

import dagger.Component
import ru.mamykin.foreignbooksreader.ReaderApp
import ru.mamykin.foreignbooksreader.di.modules.*
import ru.mamykin.foreignbooksreader.data.storage.PreferencesManager
import ru.mamykin.foreignbooksreader.presentation.about.AboutPresenter
import ru.mamykin.foreignbooksreader.presentation.bookdetails.BookDetailsPresenter
import ru.mamykin.foreignbooksreader.presentation.device.DeviceBooksPresenter
import ru.mamykin.foreignbooksreader.presentation.dropbox.DropboxBooksPresenter
import ru.mamykin.foreignbooksreader.presentation.main.MainPresenter
import ru.mamykin.foreignbooksreader.presentation.mybooks.MyBooksPresenter
import ru.mamykin.foreignbooksreader.presentation.readbook.ReadBookPresenter
import ru.mamykin.foreignbooksreader.presentation.settings.SettingsPresenter
import ru.mamykin.foreignbooksreader.presentation.store.StorePresenter
import ru.mamykin.foreignbooksreader.ui.global.BaseActivity
import ru.mamykin.foreignbooksreader.ui.mybooks.list.BooksRecyclerAdapter
import ru.mamykin.foreignbooksreader.ui.store.list.BooksStoreRecyclerAdapter
import ru.mamykin.foreignbooksreader.ui.dropbox.list.DropboxRecyclerAdapter
import ru.mamykin.foreignbooksreader.ui.devicebooks.list.FilesRecyclerAdapter
import ru.mamykin.foreignbooksreader.ui.mybooks.MyBooksFragment
import javax.inject.Singleton

/**
 * Creation date: 5/29/2017
 * Creation time: 11:39 AM
 * @author Andrey Mamykin(mamykin_av)
 */
@Component(modules = [(AppModule::class), (ApiModule::class), (DatabaseModule::class), (PreferencesModule::class), (MappersModule::class)])
@Singleton
interface AppComponent {
    fun inject(fragment: MyBooksFragment)

    fun inject(activity: BaseActivity)

    fun inject(adapter: FilesRecyclerAdapter)

    fun inject(adapter: DropboxRecyclerAdapter)

    fun inject(adapter: BooksRecyclerAdapter)

    fun inject(adapter: BooksStoreRecyclerAdapter)

    fun inject(manager: PreferencesManager)

    fun inject(presenter: BookDetailsPresenter)

    fun inject(presenter: DropboxBooksPresenter)

    fun inject(presenter: MyBooksPresenter)

    fun inject(presenter: SettingsPresenter)

    fun inject(presenter: ReadBookPresenter)

    fun inject(presenter: StorePresenter)

    fun inject(presenter: AboutPresenter)

    fun inject(presenter: DeviceBooksPresenter)

    fun inject(presenter: MainPresenter)

    fun inject(application: ReaderApp)
}